packages/amplify_authenticator/lib/src/widgets/buttons.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:amplify_authenticator/src/widgets/containers.dart';
+import 'package:flutter/material.dart';
+import 'package:amplify_authenticator/src/keys.dart';
+
+import 'package:amplify_authenticator/src/views/confirm_signup_viewmodel.dart';
+import 'package:amplify_authenticator/src/views/signin_viewmodel.dart';
+import 'package:amplify_authenticator/src/views/signup_viewmodel.dart';
+
+import 'package:amplify_authenticator/src/constants/authenticator_constants.dart';
+import 'package:amplify_authenticator/src/constants/theme_constants.dart';
+import 'package:amplify_authenticator/src/state/inherited_auth_viewmodel.dart';
+
+//Reusable buttons
+
+class SignUpButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final SignUpViewModel _authViewModel =
+        InheritedAuthViewModel.of(context)!.signUpViewModel;
+
+    return ButtonContainer(
+        callback: _authViewModel.signUp,
+        authViewModel: _authViewModel,
+        authKey: keySignUpButton,
+        text: "Sign Up");
+  }
+}
+
+class SignInButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _authViewModel = InheritedAuthViewModel.of(context)!.signInViewModel;
+
+    return ButtonContainer(
+        callback: _authViewModel.signIn,
+        authViewModel: _authViewModel,
+        authKey: keySignInButton,
+        text: "Sign In");
+  }
+}
+
+class ConfirmButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final ConfirmSignUpViewModel _authViewModel =
+        InheritedAuthViewModel.of(context)!.confirmSignUpViewModel;
+    return ButtonContainer(
+        callback: _authViewModel.confirm,
+        authViewModel: _authViewModel,
+        authKey: keyConfirmSignUpButton,
+        text: "CONFIRM");
+  }
+}
+
+class SignOutButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final SignInViewModel _authViewModel =
+        InheritedAuthViewModel.of(context)!.signInViewModel;
+    return ButtonContainer(
+        callback: _authViewModel.signOut,
+        authViewModel: _authViewModel,
+        authKey: keySignOutButton,
+        text: "Sign Out");
+  }
+}
+
+class BackToSignInButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final ConfirmSignUpViewModel _authModelView =
+        InheritedAuthViewModel.of(context)!.confirmSignUpViewModel;
+    return TextButton(
+      key: const Key(keyBackToSignInButton),
+      child: Text(
+        "Back to Sign In",
+        style: TextStyle(
+            fontSize: AuthenticatorButtonConstants.fontSize,
+            color: Theme.of(context).primaryColor != Colors.blue
+                ? Theme.of(context).primaryColor
+                : AuthenticatorColors.primary),
```

Comment:
Not sure I follow this logic

Reply:
That code is for testing out how the styling api might look like. It's gonna be replaced on the following days for the correct theming implementation.

Reply:
Sounds good 👍